lib/protocol/SqlString.js

```diff
@@ -36,6 +36,10 @@ SqlString.escape = function(val, stringifyObjects, timeZone) {
   }
 
   if (typeof val === 'object') {
+    // for the mysql point class
+    if(val.hasOwnProperty('x') && val.hasOwnProperty('y')) {
```

Comment:
We can also make this a little less risky by saying "if it has x and y _and no other properties_"